sites/partners/src/settings/PreferenceDrawer.tsx

```diff
@@ -112,44 +180,149 @@ const PreferenceDrawer = ({ drawerOpen, setDrawerOpen }: PreferenceDrawerProps)
+            <GridCell>
+              <ViewItem label={t("settings.preferenceOptOut")} className="mb-1" />
+              <FieldGroup
+                name="canYouOptOutQuestion"
+                type="radio"
+                register={register}
+                fields={[
+                  {
+                    id: YesNoAnswer.Yes,
+                    label: t("t.yes"),
+                    value: YesNoAnswer.Yes,
+                    defaultChecked: questionData === null || questionData?.optOutText !== null,
+                    dataTestId: "opt-out-question-yes",
+                  },
+                  {
+                    id: YesNoAnswer.No,
+                    label: t("t.no"),
+                    value: YesNoAnswer.No,
+                    defaultChecked: questionData && questionData?.optOutText === null,
+                    dataTestId: "opt-out-question-no",
+                  },
+                ]}
+                fieldClassName="m-0"
+                fieldGroupClassName="flex h-12 items-center"
+                dataTestId={"preference-can-you-opt-out"}
+              />
+            </GridCell>
+            {optOutQuestion === YesNoAnswer.Yes && (
+              <GridCell>
+                <ViewItem label={t("settings.preferenceOptOutLabel")}>
+                  <Field
+                    id="optOutText"
+                    name="optOutText"
+                    label={t("settings.preferenceOptOutLabel")}
+                    placeholder={t("settings.preferenceOptOutLabel")}
+                    register={register}
+                    type="text"
+                    readerOnly
+                    dataTestId={"preference-opt-out-label"}
+                    defaultValue={
+                      questionData?.optOutText ?? t("application.preferences.dontWantSingular")
+                    }
+                  />
+                </ViewItem>
+              </GridCell>
+            )}
+          </GridSection>
           <GridSection columns={3} className={"mt-4"}>
             <GridCell>
               <ViewItem label={t("settings.preferenceShowOnListing")} className="mb-1" />
               <FieldGroup
                 name="showOnListingQuestion"
                 type="radio"
                 register={register}
-                fields={yesNoOptions}
+                fields={[
+                  {
+                    id: YesNoAnswer.Yes,
+                    label: t("t.yes"),
+                    value: YesNoAnswer.Yes,
+                    defaultChecked: questionData === null || !questionData?.hideFromListing,
+                    dataTestId: "show-on-listing-question-yes",
+                  },
+                  {
+                    id: YesNoAnswer.No,
+                    label: t("t.no"),
+                    value: YesNoAnswer.No,
+                    defaultChecked: questionData?.hideFromListing,
+                    dataTestId: "show-on-listing-question-no",
+                  },
```

Comment:
When I try and update showOnListingQuestion it's updating optOutText instead, so I'm not able to update showOnListingQuestion.

Reply:
Fixed!